radixdlt-java/src/main/java/com/radixdlt/client/application/translate/ActionExecutionException.java

```diff
@@ -13,16 +13,30 @@
 	private final JsonObject errorData;
 
 	private ActionExecutionException(JsonObject errorData, List<ActionExecutionExceptionReason> reasons) {
-		super(reasons.toString());
+		super(reasons != null && !reasons.isEmpty()
+                ? reasons.toString()
+                : String.valueOf(errorData));
 
 		this.errorData = errorData;
 		this.reasons = Collections.unmodifiableList(reasons);
```

Comment:
I know this is old code, but this looks wrong to me.  It's not possible to make a List immutable post-hoc by doing this kind of thing, as the original passed-in list is still mutable and mutating it will mutate our (supposedly) unmodifiable list.

Better would be `this.reasons = ImmutableList.copyOf(reasons);`

Reply:
Would it be the Guava `com.google.common.collect.ImmutableList`?

Reply:
Yes, that is the one, otherwise it's
```
this.reasons = Collections.unmodifiableList(new ArrayList<>(reasons));
````